### PR TITLE
 Add ability to generate graphs based on correlations of sequences #25933

### LINKED ISSUE #25933: Add ability to generate graphs based on correlations of sequences
<div id="comment:0"></div>

Given a list of labeled real-valued vectors and a significance level `alpha`, generate a graph with vertices corresponding to the list's labels where two vertices are connected by an edge if the corresponding vectors have a correlation coefficient that is positive or negative at the specified significance level. An object creating the graph should support the `plot` method, which will plot the generated graph using a circular layout.

CC:  @dcoudert

Component: **graph theory**

Keywords: **sagedays@icerm**

Author: **Robert Kingan**

_Issue created by migration from https://trac.sagemath.org/ticket/25933_

### PR DESCRIPTION
**Pull request by Bruno-TT**
fixes #25933

### CODE REVIEW

File: src/sage/graphs/generators/basic.py [hunk 1]
```diff
@@ -393,6 +395,36 @@ def CompleteGraph(n):
...
+    # compare against alpha to get adjacency matrix
+    if include_anticorrelation:
+        boolean_adjacency_matrix = abs(corrs)>=alpha
+    else:
+        boolean_adjacency_matrix = corrs>=alpha
+
+    adjacency_matrix = Matrix(boolean_adjacency_matrix.astype(int))
+
+    # call graph constructor
+    return Graph(adjacency_matrix, format="adjacency_matrix", name="Correlation Graph")
```
Comment (dimpase) [inline src/sage/graphs/generators/basic.py]: check the linter error (trailing space) in this line

**Review by dimpase — CHANGES_REQUESTED** [review]

File: src/sage/graphs/generators/basic.py (hunk 1 above)
Comment (dimpase) [reply src/sage/graphs/generators/basic.py]: https://github.com/sagemath/sage/actions/runs/4117765743/jobs/7109476469

**Comment by yyyyx4** [comment]
You can use one of [a few select keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link a pull request to an issue. For example, editing the description to include "fixes #25933" will instruct GitHub to automatically close #25933 when this pull request gets merged.

**Comment by dimpase** [comment]
fixes #25933

**Comment by dimpase** [comment]
that's why tests fail:
```
sage -t --random-seed=258796353776112123915659578485470282204 sage/graphs/generators/basic.py
**********************************************************************
File "sage/graphs/generators/basic.py", line 408, in sage.graphs.generators.basic.CorrelationGraph
Failed example:
    from sage.graphs.correlation_graph import CorrelationGraph
Exception raised:
    Traceback (most recent call last):
      File "/__w/sage/sage/src/sage/doctest/forker.py", line 695, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/__w/sage/sage/src/sage/doctest/forker.py", line 1093, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.graphs.generators.basic.CorrelationGraph[0]>", line 1, in <module>
        from sage.graphs.correlation_graph import CorrelationGraph
    ModuleNotFoundError: No module named 'sage.graphs.correlation_graph'
```

**Comment by dimpase** [comment]
and linting fails due to
```
pycodestyle-minimal: commands[0] /home/runner/work/sage/sage/src> pycodestyle --select E111,E211,E271,E306,E401,E502,E701,E702,E703,W291,W391,W605,E711,E712,E713,E721,E722
./sage/graphs/generators/basic.py:427:88: W291 trailing whitespace
1       W291 trailing whitespace
```

**Review by dimpase — APPROVED** [review]

**Pull request merged**